packages/terra-section-header/src/SectionHeader.jsx

```diff
@@ -138,17 +148,23 @@ class SectionHeader extends React.Component {
     // eslint doesn't know about this and so it marks this as a lint error
     /* eslint-disable jsx-a11y/click-events-have-key-events */
     /* eslint-disable jsx-a11y/no-static-element-interactions */
+
+    if (onClick) {
+      return (
+        <Element {...attributes} className={sectionHeaderClassNames}>
+          <div role="button" aria-expanded={isOpen} tabIndex="-1" aria-label={headerText} className={cx('arrange-wrapper')}>
+            {arrangeComp}
+          </div>
+        </Element>
+      );
+    }
+
     return (
-      <Element {...attributes} onClick={onClick} className={sectionHeaderClassNames} tabIndex="0">
-        <div role="button" aria-expanded={isOpen} tabIndex="-1" aria-label={headerText} className={cx('arrange-wrapper')}>
```

Comment:
instead of two return statements can we simplify this change like below
```js
 const buttonAttributes = (onClick) ? { role: 'button', 'aria-expanded': isOpen } : undefined;

 <div {...buttonAttributes} tabIndex="-1" aria-label={headerText} className={cx('arrange-wrapper')}>
```

Reply:
Yes. But the idea was to completely remove the extra div for non-interactive section headers.

Reply:
Updated in https://github.com/cerner/terra-core/pull/3730/commits/94637158a96f84dd353cb7631dbfb8d4526750cd